src/Orleans.Runtime/Messaging/IncomingMessageAcceptor.cs

```diff
@@ -668,7 +668,7 @@ public void ProcessReceived(SocketAsyncEventArgs e)
                         {
                             // If deserialization completely failed or the message was one-way, rethrow the exception
                             // so that it can be handled at another level.
-                            if (msg?.Headers == null || msg.Direction != Message.Directions.Request)
+                            if (msg == null || msg.Direction != Message.Directions.Request)
```

Comment:
I think the behavior we want here is:

if we have a header and the message is a 'request', generate a response, log and move on.
if we have a header and the message is not a request, we log and move on.
if we have no header, we throw, which closes socket and resets buffer.

Does that seem like what we want?